### PR TITLE
Docs: Source.repeat (and friends)

### PR DESCRIPTION
**Pull request by ennru**
References #25468

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/operators/Source/repeat.md [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
-@@signature [Source.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala) { #repeat }
-
-@@@
+@apidoc[Source.repeat](Source$) { scala="#repeat[T](element:T):akka.stream.scaladsl.Source[T,akka.NotUsed]" java="#repeat(T)" }
 
 ## Description
 
-Stream a single object repeatedly
+This source emits a single element repeatedly. It never completes, so the stream will in most cases
+be limited by other means.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/stream/operators/Source/repeat.md]: Should it rather say "It never completes, if you want the stream to be finite you will need to limit it buy combining with another operator" or something along those lines?

File: akka-docs/src/main/paradox/stream/operators/Source/repeat.md [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+
+See also:
+
+* @ref:[`single`](single.md) Stream a single object once.
+* @ref:[`tick`](tick.md) A periodical repetition of an arbitrary object.
+* @ref:[`cycle`](cycle.md) Stream iterator in cycled manner.
+
+## Example
+
```
Comment (johanandren) [inline akka-docs/src/main/paradox/stream/operators/Source/repeat.md]: "In this sample we make the stream finite by combining with `take` making it complete after 17 elements:"

File: akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java [hunk 3]
```diff
@@ -626,6 +626,17 @@ public void mustRepeat() throws Exception {
...
 
+  @Test
+  public void mustRepeatForDocs() throws Exception {
+    // #repeat
+    Source<Integer, NotUsed> source = Source.repeat(42);
+    CompletionStage<List<Integer>> f = source.take(17).runWith(Sink.seq(), system);
+    // #repeat
+    final List<Integer> result = f.toCompletableFuture().get(3, TimeUnit.SECONDS);
+    assertEquals(result.size(), 17);
+    for (Integer i : result) assertEquals(i, (Integer) 42);
```
Comment (johanandren) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java]: Matter of taste but I like the non-testy ones with printlns and a comment saying what they'd print better than collecting into a seq. Not covered by regression test but more readable for the end user which is perhaps more important.

File: akka-docs/src/main/paradox/stream/operators/Source/repeat.md (hunk 1 above)
Comment (renatocaval) [reply akka-docs/src/main/paradox/stream/operators/Source/repeat.md]: if you want the stream to be finite you will need to limit it buy me a beer

File: akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java (hunk 3 above)
Comment (renatocaval) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java]: I agree with that. It also removes the need to assign it to a `f`. 

Would it be more readable if we simply write:

```java
Source
	.repeat(43)
    .take(17)
    .runWith(Sink.println(), system)
```

ps: `Sink.println()`, I made this up. I don't know if such a sink exists

Comment (renatocaval) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java]: I guess what I was looking for is something like `Sink.foreach(println)`

**Review by octonato — APPROVED** [review]

**Review by raboof — APPROVED** [review]

**Pull request merged**